### PR TITLE
Ds 472 api aggregation

### PR DESCRIPTION
**Pull request by playmakervi**
API endpoint that performs aggregate functions as well as group by functionality.

@afrasier

### CODE REVIEW

**Comment by afrasier** [comment]
Looks like there are some codeclimate fixes that need to happen.
Also, why comment out the debug code?

**Comment by bsweger** [comment]
@afrasier FYI my bad for not fixing up tests, etc. before asking @playmakervi to submit the PR. Will fix the tests and merge conflict and update.

**Comment by bsweger** [comment]
@afrasier This looks good to go for review when you get a chance!

File: usaspending_api/common/mixins.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                    aggregate=agg_function(agg_field)))
+        else:
+            # group queryset by a non-date field and aggregate
+            aggregate = (
+                queryset.annotate(item=F(group_field)).values('item').annotate(
+                    aggregate=agg_function(agg_field)))
+
+        return aggregate
+
+    def validate_request(self, params):
```
Comment (afrasier) [inline usaspending_api/common/mixins.py]: Nice validation

**Pull request merged**